src/elements/content-explorer/MetadataViewContainer.tsx

```diff
@@ -81,59 +116,92 @@ const MetadataViewContainer = ({
     columns,
     currentCollection,
     metadataTemplate,
+    onMetadataFilter,
     onSortChange: onSortChangeInternal,
+    tableProps,
     ...rest
 }: MetadataViewContainerProps) => {
+    const { formatMessage } = useIntl();
     const { items = [] } = currentCollection;
-    const { initialFilterValues: initialFilterValuesProp, onFilterSubmit: onFilterSubmitProp } = actionBarProps ?? {};
-
-    const filterGroups = React.useMemo(
-        () => [
+    const { initialFilterValues: initialFilterValuesProp, onFilterSubmit } = actionBarProps ?? {};
+
+    const newColumns = React.useMemo(() => {
+        const clonedColumns = cloneDeep(columns);
+        const hasItemNameField = clonedColumns.some((col: Column) => col.id === FIELD_ITEM_NAME);
+
+        if (!hasItemNameField) {
+            clonedColumns.unshift({
```

Comment:
the object mutation with `unshift` makes this hard to follow

Reply:
used spread instead